stix_shifter_modules/azure_log_analytics/stix_transmission/connector.py

```diff
@@ -50,39 +83,254 @@ async def create_results_connection(self, query, offset, length):
+                    response_dict['message'] = "Invalid Parameter: workspaceId"
+                else:
+                    response_dict['code'] = 400
+                    response_dict['message'] = "Invalid Query"
+            else:
+                response_dict['code'] = 500
+                response_dict['message'] = "Invalid Host/Port"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except ValueError as ex:
+            if "Invalid tenant" in str(ex):
+                response_dict['message'] = "Invalid Parameter: tenant"
+            else:
+                response_dict['message'] = "Invalid Parameter"
+            response_dict['code'] = 404
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except TimeoutError as ex:
+            response_dict['code'] = 408
+            response_dict['message'] = 'TimeoutError ' + str(ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except Exception as ex:
+            response_dict['message'] = ex
+            self.logger.error('error when getting search results: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
         return return_obj
+
+    def format_response(self, data):
+        """ Format the API responses
+        :param data: list
+        :return: list """
+        self.severity = {value: key for key, value in self.config_map["enum_supported_values"]["Severity"].items()}
+
+        for row in data:
+            copied_row = copy.deepcopy(row)
+            try:
+                # formatting security alert response
+                if row.get("Type") == "SecurityAlert":
+                    row = self.format_security_alert(row)
+                # formatting security incident response
+                elif row.get("Type") == 'SecurityIncident':
+                    row = self.format_security_incident(row)
+                # formatting security event response
+                elif row.get("Type") == "SecurityEvent":
+                    if row.get('FileHash'):
+                        row = self.find_filehash_map(row)
+            except:
+                row = copied_row
+        return data
+
+    def format_security_alert(self, security_alert):
+        """ Format the security alert response
+        :param security_alert: dict
+        :return: dict """
+
+        if security_alert.get("Techniques") == "":
+            security_alert["Techniques"] = None
+
+        if security_alert.get("ExtendedProperties"):
+            security_alert["ExtendedProperties"] = json.loads(security_alert["ExtendedProperties"])
+
+        if security_alert.get("AlertSeverity"):
+            if self.severity.get(security_alert["AlertSeverity"]):
+                security_alert["AlertSeverity"] = self.severity.get(security_alert["AlertSeverity"])
```

Comment:
the severity value translation could be a transformer. i would try using a transformer in to_stix instead of processing it here. the transformer could be reusable.

Reply:
> the severity value translation could be a transformer. i would try using a transformer in to_stix instead of processing it here. the transformer could be reusable.

Added this conversion in transformer.